modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java

```diff
@@ -573,16 +590,30 @@ public boolean stopRaftNodes(ReplicationGroupId groupId) {
 
     @Override
     public void destroyRaftNodeStorages(RaftNodeId nodeId, RaftGroupOptions groupOptions) {
-        // TODO: IGNITE-23079 - improve on what we do if it was not possible to destroy any of the storages.
+        DestroyStorageIntent intent = groupStoragesContextResolver.getIntent(nodeId, groupOptions.volatileStores());
+
+        groupStoragesDestructionIntents.saveDestroyStorageIntent(nodeId.groupId(), intent);
+
+        destroyStorage(new DestroyStorageContext(intent, groupOptions.getLogStorageFactory(), groupOptions.serverDataPath()));
+    }
+
+    private void destroyStorage(
+            DestroyStorageContext context
+    ) {
+        String nodeId = context.intent().nodeId();
+
         try {
-            String logUri = nodeId.nodeIdStringForStorage();
-            groupOptions.getLogStorageFactory().destroyLogStorage(logUri);
-        } finally {
-            Path serverDataPath = serverDataPathForNodeId(nodeId, groupOptions);
+            if (context.logStorageFactory() != null) {
+                context.logStorageFactory().destroyLogStorage(nodeId);
+            }
 
-            // This destroys both meta storage and snapshots storage as they are stored under serverDataPath.
-            IgniteUtils.deleteIfExists(serverDataPath);
+            // This destroys both meta storage and snapshots storage as they are stored under nodeDataPath.
+            IgniteUtils.deleteIfExists(getServerDataPath(context.serverDataPath(), nodeId));
```

Comment:
We might oversee an exception here. We probably need to do the following:

1. If the directory doesn't exist, we do nothing
2. Otherwise, we try to delete it
3. If an exception happens, we rethrow it